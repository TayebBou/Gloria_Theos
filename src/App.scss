.loader {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
  background-color: rgb(19, 21, 22);
  color: white;
  z-index: 1001;

  &::first-child {
    animation: fadeInOut 2s infinite;
  }
}

@keyframes fadeInOut {
  0% {
    opacity: 0;
  }
  50% {
    opacity: 1;
  }
  100% {
    opacity: 0;
  }
}
