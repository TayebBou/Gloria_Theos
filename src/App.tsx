import { OrbitControls, Stats, useGLTF, Sphere } from '@react-three/drei';
import { Canvas } from '@react-three/fiber';
import { FC } from 'react';
import Fireflies from './Fireflies/Fireflies';
import { Mesh } from 'three';

const App: FC = () => {
  const gltf = useGLTF('/models/scene.glb');

  gltf?.scene?.traverse((model) => {
    if (model instanceof Mesh && model.isMesh) {
      if (model.name !== 'Plane') model.castShadow = true;
      model.receiveShadow = true;
    }
  });

  return (
    <Canvas camera={{ position: [4.68, 2.012, 1.183] }} shadows>
      <directionalLight position={[4.49, 4.49, 3.44]} intensity={5} castShadow>
        <Sphere args={[0.25]} />
      </directionalLight>
      <primitive object={gltf?.scene} />
      <Fireflies />
      <OrbitControls
        target={[0, 1, 0]}
        minPolarAngle={0} // Minimum vertical angle in radians (0 radians = looking straight ahead)
        maxPolarAngle={Math.PI / 2} // Maximum vertical angle in radians (PI/2 radians = looking straight down)
        minDistance={1} // Minimum zoom distance
        maxDistance={6} // Maximum zoom distance
      />
      <Stats />
    </Canvas>
  );
};

export default App;
