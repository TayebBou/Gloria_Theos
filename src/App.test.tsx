import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import App from './App';

// Mock useGLTF hook
jest.mock('@react-three/drei', () => ({
  OrbitControls: () => null,
  Stats: () => null,
  useGLTF: jest.fn(() => ({
    scene: {
      traverse: jest.fn((fn: (object: any) => void) => {
        // Mock a simple object with the traverse method
        const mockObject = {
          isMesh: true,
          name: 'SomeMesh',
          castShadow: false,
          receiveShadow: false,
        };
        fn(mockObject);
      }),
    },
  })),
  Sphere: () => null,
}));

// Mock Canvas to prevent issues during tests
jest.mock('@react-three/fiber', () => ({
  Canvas: ({ children }: { children: React.ReactNode }) => <>{children}</>,
  useFrame: jest.fn(), // Mock useFrame as a jest function
}));

describe('App Component', () => {
  test('renders the App component and its elements', () => {
    const { container } = render(<App />);

    // Verify Canvas is rendered
    expect(container).toMatchSnapshot();
  });

  test('uses the useGLTF hook', () => {
    const { useGLTF } = require('@react-three/drei');
    const useGLTFMock = useGLTF as jest.Mock;

    render(<App />);

    // Check if useGLTF is called with the correct argument
    expect(useGLTFMock).toHaveBeenCalledWith('/models/scene.glb');
  });
});
