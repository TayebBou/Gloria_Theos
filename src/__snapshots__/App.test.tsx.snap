// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component renders the App component and its elements 1`] = `
<div>
  <directionallight
    intensity="5"
    position="4.49,4.49,3.44"
  />
  <primitive />
  <mesh
    position="-0.5827328389883091,-2.101486145820572,-0.675403737392994"
  >
    <meshstandardmaterial
      color="white"
      emissive="white"
      emissiveintensity="1"
    />
  </mesh>
  <pointlight
    color="white"
    decay="2"
    distance="2"
    intensity="2.5"
  />
  <mesh
    position="-1.1526619229112862,-2.6307258211277746,0.03594499992407574"
  >
    <meshstandardmaterial
      color="white"
      emissive="white"
      emissiveintensity="1"
    />
  </mesh>
  <pointlight
    color="white"
    decay="2"
    distance="2"
    intensity="2.5"
  />
  <mesh
    position="2.828942046567257,2.568252173270125,2.435636721915965"
  >
    <meshstandardmaterial
      color="white"
      emissive="white"
      emissiveintensity="1"
    />
  </mesh>
  <pointlight
    color="white"
    decay="2"
    distance="2"
    intensity="2.5"
  />
  <mesh
    position="0.8188514665363531,-0.09087840408793735,1.0435977794612108"
  >
    <meshstandardmaterial
      color="white"
      emissive="white"
      emissiveintensity="1"
    />
  </mesh>
  <pointlight
    color="white"
    decay="2"
    distance="2"
    intensity="2.5"
  />
  <mesh
    position="-2.8785311824517055,1.1782233696347562,-1.5709701453111498"
  >
    <meshstandardmaterial
      color="white"
      emissive="white"
      emissiveintensity="1"
    />
  </mesh>
  <pointlight
    color="white"
    decay="2"
    distance="2"
    intensity="2.5"
  />
  <mesh
    position="0.542440991855826,1.4708905092496969,2.3123250919549156"
  >
    <meshstandardmaterial
      color="white"
      emissive="white"
      emissiveintensity="1"
    />
  </mesh>
  <pointlight
    color="white"
    decay="2"
    distance="2"
    intensity="2.5"
  />
  <mesh
    position="2.684293493864188,-1.0145374069559443,-1.8972962836591831"
  >
    <meshstandardmaterial
      color="white"
      emissive="white"
      emissiveintensity="1"
    />
  </mesh>
  <pointlight
    color="white"
    decay="2"
    distance="2"
    intensity="2.5"
  />
  <mesh
    position="1.341694335506284,2.0349729203837397,-0.4735751036044653"
  >
    <meshstandardmaterial
      color="white"
      emissive="white"
      emissiveintensity="1"
    />
  </mesh>
  <pointlight
    color="white"
    decay="2"
    distance="2"
    intensity="2.5"
  />
  <mesh
    position="-2.4726835670961718,0.6722504320328597,2.77571474596725"
  >
    <meshstandardmaterial
      color="white"
      emissive="white"
      emissiveintensity="1"
    />
  </mesh>
  <pointlight
    color="white"
    decay="2"
    distance="2"
    intensity="2.5"
  />
  <mesh
    position="0.5523861157948122,0.015024452793205967,-1.0303006677445774"
  >
    <meshstandardmaterial
      color="white"
      emissive="white"
      emissiveintensity="1"
    />
  </mesh>
  <pointlight
    color="white"
    decay="2"
    distance="2"
    intensity="2.5"
  />
  <mesh
    position="1.365428850888705,-1.1103545095642766,-0.3407473983126472"
  >
    <meshstandardmaterial
      color="white"
      emissive="white"
      emissiveintensity="1"
    />
  </mesh>
  <pointlight
    color="white"
    decay="2"
    distance="2"
    intensity="2.5"
  />
  <mesh
    position="-1.1932259532225276,1.5583967170131898,-2.8506141643785865"
  >
    <meshstandardmaterial
      color="white"
      emissive="white"
      emissiveintensity="1"
    />
  </mesh>
  <pointlight
    color="white"
    decay="2"
    distance="2"
    intensity="2.5"
  />
  <mesh
    position="0.35932656172125155,-0.4045210376931303,1.4941601999856218"
  >
    <meshstandardmaterial
      color="white"
      emissive="white"
      emissiveintensity="1"
    />
  </mesh>
  <pointlight
    color="white"
    decay="2"
    distance="2"
    intensity="2.5"
  />
  <mesh
    position="-2.587540554520917,-2.735347740417753,0.7547922089346786"
  >
    <meshstandardmaterial
      color="white"
      emissive="white"
      emissiveintensity="1"
    />
  </mesh>
  <pointlight
    color="white"
    decay="2"
    distance="2"
    intensity="2.5"
  />
  <mesh
    position="-2.072141623828484,2.182642549725452,2.5572238629965676"
  >
    <meshstandardmaterial
      color="white"
      emissive="white"
      emissiveintensity="1"
    />
  </mesh>
  <pointlight
    color="white"
    decay="2"
    distance="2"
    intensity="2.5"
  />
  <mesh
    position="-1.5226566843492693,1.2732753405157418,1.8858842957162594"
  >
    <meshstandardmaterial
      color="white"
      emissive="white"
      emissiveintensity="1"
    />
  </mesh>
  <pointlight
    color="white"
    decay="2"
    distance="2"
    intensity="2.5"
  />
  <mesh
    position="-2.2953050821224106,1.4868568418628634,2.7394347195613693"
  >
    <meshstandardmaterial
      color="white"
      emissive="white"
      emissiveintensity="1"
    />
  </mesh>
  <pointlight
    color="white"
    decay="2"
    distance="2"
    intensity="2.5"
  />
  <mesh
    position="2.2185509044079224,2.1352285158935524,-0.3027974628342771"
  >
    <meshstandardmaterial
      color="white"
      emissive="white"
      emissiveintensity="1"
    />
  </mesh>
  <pointlight
    color="white"
    decay="2"
    distance="2"
    intensity="2.5"
  />
  <mesh
    position="2.7583567976460377,-0.6902137382409461,-2.391405660309912"
  >
    <meshstandardmaterial
      color="white"
      emissive="white"
      emissiveintensity="1"
    />
  </mesh>
  <pointlight
    color="white"
    decay="2"
    distance="2"
    intensity="2.5"
  />
  <mesh
    position="-0.7737014940920655,-0.013598044530601427,-0.8533405233033489"
  >
    <meshstandardmaterial
      color="white"
      emissive="white"
      emissiveintensity="1"
    />
  </mesh>
  <pointlight
    color="white"
    decay="2"
    distance="2"
    intensity="2.5"
  />
</div>
`;
